File: src/main/kotlin/magicbook/gtlitecore/api/recipe/builder/QuantumForceTransformerRecipeBuilder.java
package magicbook.gtlitecore.api.recipe.builder;

import gregtech.api.recipes.Recipe;
import gregtech.api.recipes.RecipeBuilder;
import gregtech.api.recipes.RecipeMap;
import gregtech.api.util.EnumValidationResult;
import magicbook.gtlitecore.api.recipe.property.QuantumForceTransformerTieredProperty;
import magicbook.gtlitecore.api.utils.GTLiteLog;
import org.apache.commons.lang3.builder.ToStringBuilder;
import org.jetbrains.annotations.NotNull;

public class QuantumForceTransformerRecipeBuilder extends RecipeBuilder<QuantumForceTransformerRecipeBuilder>
{

    public QuantumForceTransformerRecipeBuilder() {}

    public QuantumForceTransformerRecipeBuilder(Recipe recipe,
                                                RecipeMap<QuantumForceTransformerRecipeBuilder> recipeMap)
    {
        super(recipe, recipeMap);
    }

    public QuantumForceTransformerRecipeBuilder(RecipeBuilder<QuantumForceTransformerRecipeBuilder> recipeBuilder)
    {
        super(recipeBuilder);
    }

    @Override
    public QuantumForceTransformerRecipeBuilder copy()
    {
        return new QuantumForceTransformerRecipeBuilder(this);
    }

    public int getTier()
    {
        return this.recipePropertyStorage == null ? 0 : this.recipePropertyStorage.get(QuantumForceTransformerTieredProperty.getInstance(), 0);
    }

    @Override
    public boolean applyPropertyCT(@NotNull String key, @NotNull Object value)
    {
        if (key.equals(QuantumForceTransformerTieredProperty.KEY))
        {
            this.tier(((Number) value).intValue());
            return true;
        }
        return super.applyPropertyCT(key, value);
    }

    public QuantumForceTransformerRecipeBuilder tier(int tier)
    {
        if (tier <= 0)
        {
            GTLiteLog.logger.error("Tier cannot be less than or equal to 0", new IllegalArgumentException());
            this.recipeStatus = EnumValidationResult.INVALID;
        }
        this.applyProperty(QuantumForceTransformerTieredProperty.getInstance(), tier);
        return this;
    }

    @Override
    public String toString()
    {
        return new ToStringBuilder(this)
                .appendSuper(super.toString())
                .append(QuantumForceTransformerTieredProperty.getInstance().getKey(), this.getTier())
                .toString();
    }

}

File: src/main/kotlin/gregtechlite/gtlitecore/api/recipe/builder/QuantumForceTransformerRecipeBuilder.kt
package gregtechlite.gtlitecore.api.recipe.builder

import gregtech.api.recipes.Recipe
import gregtech.api.recipes.RecipeBuilder
import gregtech.api.recipes.RecipeMap
import gregtech.api.util.EnumValidationResult
import gregtechlite.gtlitecore.api.GTLiteLog
import gregtechlite.gtlitecore.api.recipe.GTLiteRecipeProperties
import gregtechlite.gtlitecore.api.util.buildToString

class QuantumForceTransformerRecipeBuilder : RecipeBuilder<QuantumForceTransformerRecipeBuilder>
{
    val tier = recipePropertyStorage?.let { recipePropertyStorage.get(GTLiteRecipeProperties.QUANTUM_FORCE_TRANSFORMER_TIER, 0) } ?: 0

    constructor()

    @Suppress("unused")
    constructor(recipe: Recipe, recipeMap: RecipeMap<QuantumForceTransformerRecipeBuilder>) : super(recipe, recipeMap)

    constructor(recipeBuilder: RecipeBuilder<QuantumForceTransformerRecipeBuilder>) : super(recipeBuilder)

    override fun copy(): QuantumForceTransformerRecipeBuilder = QuantumForceTransformerRecipeBuilder(this)

    override fun applyPropertyCT(key: String, value: Any): Boolean
    {
        if (key == GTLiteRecipeProperties.QUANTUM_FORCE_TRANSFORMER_TIER.key)
        {
            tier((value as Number).toInt())
            return true
        }
        return super.applyPropertyCT(key, value)
    }

    fun tier(tier: Int): QuantumForceTransformerRecipeBuilder = apply {
        if (tier <= 0)
        {
            GTLiteLog.logger.error("Tier cannot be less than or equal to 0", IllegalArgumentException())
            recipeStatus = EnumValidationResult.INVALID
        }
        applyProperty(GTLiteRecipeProperties.QUANTUM_FORCE_TRANSFORMER_TIER, tier)
    }

    override fun toString(): String = buildToString(this) {
        appendSuper(super.toString())
        append(GTLiteRecipeProperties.QUANTUM_FORCE_TRANSFORMER_TIER.key, tier)
    }
}
